lib/Dialect/Torch/Transforms/DecomposeComplexOps.cpp

```diff
@@ -4416,6 +4416,53 @@ class DecomposeAtenNewEmptyStridedOp
 };
 } // namespace
 
+namespace {
+class DecomposeAtenEmptyStridedOp
```

Comment:
This pattern is almost identical to the pattern for `DecomposeAtenNewEmptyStridedOp`. We should avoid copying and pasting code when possible to make maintenance easier. Here, we can avoid repetition by having `AtenNewEmptyStridedOp` decompose into `atenEmptyStridedOp`.

Reply:
@brucekimrokcmu can you send an NFC (non-functional change) PR when you have time to address Ramiro's comments on code organization?

Reply:
@ramiro050 Hi Ramiro, 
I was initially thinking if I could simply decompose `AtenEmptyStridedOp` into `AtenNewEmptyStridedOp` to avoid any repetition. Could you look at my approach below and give me feed back how it can be improved?

Here's how `new_empty_strided` is decomposed:
`AtenNewEmptyStridedOp` -> `AtenNewEmptyOp` -> `AtenEmptyMemoryFormatOp`
Here, `self` tensor is passed, and is also required by `AtenNewEmptyOp`, according table gen.
 
On the other hand `AtenEmptyStridedOp` does not pass a tensor, and `AtenEmptyMemoryFormatOp` requires no `self` in table gen. So it cannot be decomposed into `AtenNewEmptyOp`, but would have to be decomposed into `AtenEmptyMemoryFormatOp` directly. 

So here, it is decomposed as the following: 
`AtenEmptyStridedOp` -> `AtenEmptyMemoryFormatOp` 

Please let me know your thoughts and I will make NFC PR accordingly.

Reply:
Yeah, all paths lead to `AtenEmptyMemoryFormatOp`. The `AtenNew...` ops have [value semantics in the table gen generated](https://github.com/llvm/torch-mlir/blob/40913a36c297239523863867f2e9b8639176993f/include/torch-mlir/Dialect/Torch/IR/GeneratedTorchOps.td#L8227), which means that the `self` argument is not modified, so we can drop it (it is only used to [set `dtype` and `device` args](https://pytorch.org/docs/stable/generated/torch.Tensor.new_empty.html) if they're set to `None` in `AtenNew...`. `device` can be ignored, since there is no concept of device in torch-mlir. `dtype` can be handled [like this](https://github.com/brucekimrokcmu/torch-mlir/blob/dcd56980c4f4cff2d68dad231f83b29ba4b3e8c8/lib/Dialect/Torch/Transforms/DecomposeComplexOps.cpp#L3288-L3297)).

The flow to avoid too much code repetition should be:
- `AtenEmptyStridedOp` -> `AtenEmptyMemoryFormatOp`
- `AtenNewEmptyStridedOp` (dropping `self` and properly setting `dtype` arg) -> `AtenEmptyStridedOp`

Reply:
That sounds good! I can work on those after I finish the priorities I have :) Thanks for the feedback @ramiro050

Reply:
Sure!